src/delegation.rs

```diff
@@ -82,8 +248,8 @@ impl<DID: Did, V: varsig::Header<C>, C: Codec> Delegation<DID, V, C> {
     }
 
     /// Retrive the `subject` of a [`Delegation`]
-    pub fn subject(&self) -> Option<&DID> {
-        self.payload.subject.as_ref()
+    pub fn subject(&self) -> &Subject<DID> {
+        &self.payload.subject
     }
```

Comment:
If we change this, then a petition to keep it consistent with how `via()`, `not_before()` and `expiration()` work.

Reply:
How's that?

Reply:
Hm. Given that there are two changes at the same time: From `Option` -> `Subject` and from `T<&S>` to `&T<S>`.
Can we implement `Subject::as_ref(&self) -> Subject<&T>` and make this return `Subject<&DID>`?